File: PilluleBoxApp/PilluleBox/app/src/main/java/Models/ScheduleModes/BasicMode.java
package Models.ScheduleModes;

import java.io.Serializable;
import java.sql.Time;

public class BasicMode implements Serializable {
    private int id;
    private String medicineName;
    private Time morningStartTime;
    private Time morningEndTime;
    private Time afternoonStartTime;
    private Time afternoonEndTime;
    private Time nightStartTime;
    private Time nightEndTime;

    public BasicMode(String medicineName,
                     Time morningStartTime, Time morningEndTime,
                     Time afternoonStartTime, Time afternoonEndTime,
                     Time nightStartTime, Time nightEndTime) {
        this.medicineName = medicineName;
        this.morningStartTime = morningStartTime;
        this.morningEndTime = morningEndTime;
        this.afternoonStartTime = afternoonStartTime;
        this.afternoonEndTime = afternoonEndTime;
        this.nightStartTime = nightStartTime;
        this.nightEndTime = nightEndTime;
    }

    public BasicMode(int id, String medicineName,
                     Time morningStartTime, Time morningEndTime,
                     Time afternoonStartTime, Time afternoonEndTime,
                     Time nightStartTime, Time nightEndTime) {
        this.id = id;
        this.medicineName = medicineName;
        this.morningStartTime = morningStartTime;
        this.morningEndTime = morningEndTime;
        this.afternoonStartTime = afternoonStartTime;
        this.afternoonEndTime = afternoonEndTime;
        this.nightStartTime = nightStartTime;
        this.nightEndTime = nightEndTime;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getMedicineName() { return medicineName; }
    public void setMedicineName(String medicineName) { this.medicineName = medicineName; }

    public Time getMorningStartTime() { return morningStartTime; }
    public void setMorningStartTime(Time morningStartTime) { this.morningStartTime = morningStartTime; }

    public Time getMorningEndTime() { return morningEndTime; }
    public void setMorningEndTime(Time morningEndTime) { this.morningEndTime = morningEndTime; }

    public Time getAfternoonStartTime() { return afternoonStartTime; }
    public void setAfternoonStartTime(Time afternoonStartTime) { this.afternoonStartTime = afternoonStartTime; }

    public Time getAfternoonEndTime() { return afternoonEndTime; }
    public void setAfternoonEndTime(Time afternoonEndTime) { this.afternoonEndTime = afternoonEndTime; }

    public Time getNightStartTime() { return nightStartTime; }
    public void setNightStartTime(Time nightStartTime) { this.nightStartTime = nightStartTime; }

    public Time getNightEndTime() { return nightEndTime; }
    public void setNightEndTime(Time nightEndTime) { this.nightEndTime = nightEndTime; }
}

File: PilluleBoxESP32/PilluleBox/BasicMode.cpp
#include "pillulebox.h"

BasicMode::BasicMode(int id) {
  this->id = id;
}

BasicMode::~BasicMode() {}

int BasicMode::getId() const {
  return id;
}

const char* BasicMode::getMedicineName() const {
  return medicine_name;
}

tm BasicMode::getMorningStartTime() const {
  return morning_start_time;
}

tm BasicMode::getMorningEndTime() const {
  return morning_end_time;
}

tm BasicMode::getAfternoonStartTime() const {
  return afternoon_start_time;
}

tm BasicMode::getAfternoonEndTime() const {
  return afternoon_end_time;
}

tm BasicMode::getNightStartTime() const {
  return night_start_time;
}

tm BasicMode::getNightEndTime() const {
  return night_end_time;
}

void BasicMode::setId(int id) {
  this->id = id;
}

void BasicMode::setMedicineName(const char* name) {
  strncpy(medicine_name, name, sizeof(medicine_name) - 1);
  medicine_name[sizeof(medicine_name) - 1] = '\0';  // Asegura la terminación nula
}

void BasicMode::setMorningStartTime(const tm& time) {
  morning_start_time = time;
}

void BasicMode::setMorningEndTime(const tm& time) {
  morning_end_time = time;
}

void BasicMode::setAfternoonStartTime(const tm& time) {
  afternoon_start_time = time;
}

void BasicMode::setAfternoonEndTime(const tm& time) {
  afternoon_end_time = time;
}

void BasicMode::setNightStartTime(const tm& time) {
  night_start_time = time;
}

void BasicMode::setNightEndTime(const tm& time) {
  night_end_time = time;
}